public/hooks/use_chat_actions.tsx

```diff
@@ -174,8 +182,28 @@ export const useChatActions = (): AssistantActions => {
         if (abortController.signal.aborted) {
           return;
         }
+        /**
+         * Remove the regenerated interaction & message.
+         * In implementation of Agent framework, it will generate a new interactionId
+         * so need to remove the staled interaction in Frontend manually.
+         */
+        const findRegeratedMessageIndex = chatState.messages.findLastIndex(
+          (message) => message.type === 'input'
+        );
+        if (findRegeratedMessageIndex > -1) {
+          chatStateDispatch({
+            type: 'receive',
+            payload: {
+              messages: chatState.messages.slice(0, findRegeratedMessageIndex),
```

Comment:
Do we need to filter empty messageId items here?

Reply:
Actually the empty messageId items will be sliced out in this line, but it has no harm to add a filter here. Done.